syntax = "proto3";

package im;

import "common.proto";

service AuthService {
    rpc Register(RegisterRequest) returns (RegisterResponse);
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc Logout(LogoutRequest) returns (LogoutResponse);
    rpc RefreshToken(RefreshTokenRequest) returns (RefreshTokenResponse);
    rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string phone = 3;
    string email = 4;
    string nickname = 5;
}

message RegisterResponse {
    bool success = 1;
    Error error = 2;
    int64 user_id = 3;
}

message LoginRequest {
    string username = 1;
    string password = 2;
    string device_id = 3;
    string device_type = 4;
    string client_version = 5;
}

message LoginResponse {
    bool success = 1;
    Error error = 2;
    int64 user_id = 3;
    string token = 4;
    int64 expire_at = 5;
}

message LogoutRequest {
    int64 user_id = 1;
    string device_id = 2;
}

message LogoutResponse {
    bool success = 1;
    Error error = 2;
}

message RefreshTokenRequest {
    string refresh_token = 1;
}

message RefreshTokenResponse {
    bool success = 1;
    Error error = 2;
    string token = 3;
    int64 expire_at = 4;
}

message ValidateTokenRequest {
    string token = 1;
}

message ValidateTokenResponse {
    bool valid = 1;
    int64 user_id = 2;
}
